### DISCUSSION TITLE
Help: Implementation of leap.nvim

### DISCUSSION THREAD

**Discussion by ChiefMateStarbuck** [helix-editor/helix#12352] [General]
Helix already has a text jumper in the form of `goto_word`. However, it is amp based. 

I would like to implement a [leap.nvim](https://github.com/ggandor/leap.nvim) jumper. leap style jumping can be less overhead for some users.

Is there any advice the community can give on implementing this? It looks like there are some [PRs](https://github.com/helix-editor/helix/pull/8875) I can piggy back off of. 

Thank you in advance.

**Comment by ChiefMateStarbuck** [comment]
So it turns out this has already been implemented but was rejected about last year time. Would anyone have the link for the PR?

**Comment by TornaxO7** [comment]
May I ask what the difference is between your suggesstion and `gw`?

**Reply by ChiefMateStarbuck** [reply]
hello @TornaxO7,

"gw" is amp based. It uses two random letters for navigation. leap.nvim use the first two letters of the desired symbol plus one random letter for navigation.

leap requires three keystrokes for navigation while, while the amp style "gw" uses two. However, since the first two letters of leap.nvim navigation are the first two letters of the desired symbol, you don't have to think really hard about them. You only have to "think" about one random character in leap.nvim as a opposed to two in amp based "gw". Additionally, that one single random letter is usually easily accessible (i.e. g,f,d,s).

This is a small differences definitely. This can be seen as personal choice. I find leap.nvim more ergonomic. I can navigate faster with less "thinking".

**Comment by kirawi** [comment]
https://github.com/helix-editor/helix/pull/5340 ?

**Reply by ChiefMateStarbuck** [reply]
Yes this looks like it. Thank you.

**Comment by sullyj3** [comment]
I'm also interested in this. Label based jumps don't cut it for me, the UX of having to read the label first to type it is too disruptive. 

Is this something that could be implemented in a Steel plugin once that drops?
